tests/test_call_policies.py

```diff
@@ -100,6 +100,7 @@ class Derived(Parent):
     with capture:
         del p, lst
         pytest.gc_collect()
+        pytest.gc_collect()    # Needed to make PyPy free the child
```

Comment:
There's a `ConstructorStats.detail_reg_inst()` you can call here which handles the GC collection (including doing it twice for PyPy), and also returns the number of currently tracked instances.  Aside from not needing the manual GC calls at all (and not needing to worry about invoking it twice under PyPy), it might be a useful check here, as well, to make sure that another object hasn't been created/copied somewhere along the line:

```Python
    n_inst = ConstructorStats.detail_reg_inst()
    # ... create parent, child
    assert ConstructorStats.detail_reg_inst() == n_inst + 2
    # ... destroy
    assert ConstructorStats.detail_reg_inst() == n_inst
```